src/pages/kandidat/Kandidatdetaljer.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect } from 'react';
+import { RouteComponentProps, withRouter } from 'react-router';
+import NavFrontendSpinner from 'nav-frontend-spinner';
+
+import { AppRoute, MatchProps } from '../../utils/paths';
+import { formaterFnr } from '../før-du-begynner/fnr-input/fnrUtils';
+import { hentKandidat, hentSkrivetilgang } from '../../api/api';
+import bemHelper from '../../utils/bemHelper';
+import EndreKandidat from './endre-kandidat/EndreKandidat';
+import EndreKandidatKnapp from './endre-kandidat/EndreKandidatKnapp';
+import Kandidat from '../../types/Kandidat';
+import Placeholderboks from '../../components/placeholder-boks/PlaceholderBoks';
+import RouteBanner from '../../components/route-banner/RouteBanner';
+import SeKandidat from './se-kandidat/SeKandidat';
+import Tilbake from '../../components/tilbake/Tilbake';
+import './kandidatdetaljer.less';
+
+const cls = bemHelper('kandidatdetaljer');
+
+interface OwnProps {
+    endre: boolean;
+}
+
+type Props = OwnProps & RouteComponentProps<MatchProps>;
+
+const Kandidatdetaljer = (props: Props) => {
+    const { match, endre } = props;
+
+    const [kandidat, setKandidat] = useState<Kandidat | undefined>(undefined);
+    const [feilmelding, setFeilmelding] = useState<string | undefined>(undefined);
+    const [harSkrivetilgang, settSkrivetilgang] = useState<boolean>(false);
+
+    const fnr = match.params.fnr;
+
+    const hentKandidatFraUrl = async () => {
+        try {
+            const kandidat = await hentKandidat(fnr);
+            setKandidat(kandidat);
+        } catch (error) {
+            setFeilmelding(
+                'Du har enten ikke tilgang til denne kandidaten eller så finnes ikke kandidaten i systemet'
+            );
+        }
+    };
+
+    const sjekkSkrivetilgang = async () => {
+        const harTilgang = await hentSkrivetilgang(fnr);
+        settSkrivetilgang(harTilgang);
+    };
+
+    useEffect(() => {
```

Comment:
Det er ofte greit at funksjoner med minst implementasjonsdetaljer kommer tidlig i en fil. Når man kommer inn i klassen skjønner man fort hovedlogikken, så kan man bla seg videre nedover og lese implementasjonen til funksjonene om man ønsker det. Så kanskje ha `useEffect` før de andre to funksjonene :)